index.js

```diff
@@ -323,49 +328,51 @@ function handleResponses(expressApp,
    * because this comes before it.
    */
   app = expressApp;
-  swaggerUiServePath = options.swaggerUiServePath || 'api-docs';
+  swaggerUiServePath = options.swaggerUiServePath || DEFAULT_SWAGGER_UI_SERVE_PATH;
   predefinedSpec = options.predefinedSpec || {};
   const { specOutputPath, writeIntervalMs } = options;
 
   updateDefinitionsSpec(options.mongooseModels);
   updateTagsSpec(options.tags || options.mongooseModels);
   
-  /** middleware to handle RESPONSES */
-  app.use((req, res, next) => {
-    try {
-      const methodAndPathKey = getMethod(req);
-      if (methodAndPathKey && methodAndPathKey.method) {
-        processors.processResponse(res, methodAndPathKey.method);
-      }
-
-      const ts = new Date().getTime();
-      if (firstResponseProcessing || specOutputPath && ts - lastRecordTime > writeIntervalMs) {
-        firstResponseProcessing = false;
-        lastRecordTime = ts;
-
-        fs.writeFileSync(specOutputPath, JSON.stringify(spec, null, 2), 'utf8');
-
-        convertOpenApiVersionToV3(spec, (err, specV3) => {
-          if (!err) {
-            const parsedSpecOutputPath = path.parse(specOutputPath);
-            const {name, ext} = parsedSpecOutputPath;
-            parsedSpecOutputPath.base = name.concat('_').concat(versions.OPEN_API_V3).concat(ext);
-            
-            const v3Path = path.format(parsedSpecOutputPath);
-            
-            fs.writeFileSync(v3Path, JSON.stringify(specV3), 'utf8');
-          }
-          /** TODO - Log that open api v3 could not be generated */
-        });  
-
+  if (!ignoredNodeEnvironments.includes(process.env.NODE_ENV)) {
```

Comment:
instead of

```js
if (!ignoredNodeEnvironments.includes(process.env.NODE_ENV)) {
   do_stuff();
}
```

go with

```js
if (ignoredNodeEnvironments.includes(process.env.NODE_ENV)) {
	return;
}

do_stuff();
```

this way the indentation and line-blame won't need to change, and will be generally easier to follow.

Reply:
Yeah, I love early return. But test coverage was not enough in that case because of unreachable lines.
The way to test would be to force the `process.env.NODE_ENV` to production in any test, which is unfair.

Reply:
Let's use Guard Clauses (early return) pattern?

BTW didn't notice any new tests or changes in existing tests related to that and following `if` conditions.

Reply:
Just fixed early return + Included integration test (sorry for the callback hell, promisify tests is another thing to refactor)

Reply:
excellent